javaslang/src/test/java/javaslang/collection/TreeMapTest.java

```diff
@@ -126,6 +125,12 @@ public void shouldWrapMap() {
         assertThat(TreeMap.ofAll(source)).isEqualTo(emptyIntInt().put(1, 2).put(3, 4));
     }
 
+    @Test
+    public void shouldCreateKeyComparatorForJavaUtilMap() {
+        final TreeMap<String, String> map = TreeMap.ofAll(Collections.singletonMap("key", "value"));
+        assertThat(map.keyComparator()).isNotNull();
```

Comment:
We could ensure that the natural Tuple2 comparator is used, namely the component-wise comparator.

An additional test for this would create `TreeMap.of(Tuple.of("b", "b"), Tuple.of("a", "b"), Tuple.of("b", "a")))`.

Don't know what's the best way to check the validity. Maybe `assertThat(treeMap.toList()).isEqualTo(List.of(Tuple.of("a", "b"), Tuple.of("b", "a"), Tuple.of("b", "b")))`?